extensions/sql-migration/src/dashboard/sqlServerDashboard.ts

```diff
@@ -269,40 +275,78 @@ export class DashboardWidget {
-								if (v === loc.YES) {
-									const cutoverDialogModel = new MigrationCutoverDialogModel(
-										await MigrationLocalStorage.getMigrationServiceContext(),
-										migration!);
-									await cutoverDialogModel.fetchStatus();
-									await cutoverDialogModel.cancelMigration();
-
-									if (cutoverDialogModel.CancelMigrationError) {
-										void vscode.window.showErrorMessage(loc.MIGRATION_CANNOT_CANCEL);
-										logError(TelemetryViews.MigrationsTab, MenuCommands.CancelMigration, cutoverDialogModel.CancelMigrationError);
+		this._context.subscriptions.push(
+			vscode.commands.registerCommand(
+				MenuCommands.CancelMigration,
+				async (args: MenuCommandArgs) => {
+					try {
+						await this.clearError(args.connectionId);
+						const migration = await this._getMigrationById(args.migrationId, args.migrationOperationId);
+						if (migration && canCancelMigration(migration)) {
+							void vscode.window
+								.showInformationMessage(loc.CANCEL_MIGRATION_CONFIRMATION, loc.YES, loc.NO)
+								.then(async (v) => {
+									if (v === loc.YES) {
+										const cutoverDialogModel = new MigrationCutoverDialogModel(
+											await MigrationLocalStorage.getMigrationServiceContext(),
+											migration!);
+										await cutoverDialogModel.fetchStatus();
+										await cutoverDialogModel.cancelMigration();
+
+										if (cutoverDialogModel.CancelMigrationError) {
+											void vscode.window.showErrorMessage(loc.MIGRATION_CANNOT_CANCEL);
+											logError(TelemetryViews.MigrationsTab, MenuCommands.CancelMigration, cutoverDialogModel.CancelMigrationError);
+										}
 									}
-								}
-							});
-					} else {
-						await vscode.window.showInformationMessage(loc.MIGRATION_CANNOT_CANCEL);
+								});
+						} else {
+							await vscode.window.showInformationMessage(loc.MIGRATION_CANNOT_CANCEL);
+						}
+					} catch (e) {
+						await this.showError(
+							args.connectionId,
+							loc.MIGRATION_CANCELLATION_ERROR,
+							loc.MIGRATION_CANCELLATION_ERROR,
+							e.message);
+						logError(TelemetryViews.MigrationsTab, MenuCommands.CancelMigration, e);
 					}
-				} catch (e) {
-					await this.showError(
-						args.connectionId,
-						loc.MIGRATION_CANCELLATION_ERROR,
-						loc.MIGRATION_CANCELLATION_ERROR,
-						e.message);
-					logError(TelemetryViews.MigrationsTab, MenuCommands.CancelMigration, e);
-				}
-			}));
+				}));
+
+		this._context.subscriptions.push(
+			vscode.commands.registerCommand(
+				MenuCommands.DeleteMigration,
+				async (args: MenuCommandArgs) => {
+					await this.clearError(args.connectionId);
+					try {
+						const service = await MigrationLocalStorage.getMigrationServiceContext();
+						const migration = await this._getMigrationById(args.migrationId, args.migrationOperationId);
+						if (service && (
+							(migration && canDeleteMigration(migration)) ||
+							(migration === undefined && args.migrationId?.length > 0))) {
```

Comment:
this part is getting a bit messy to read, maybe move to helper fuction? also can you add comment why we want to fallback even if migration is undefined?